### ISSUE TITLE
Camunda Modeler Keyboard Shortcuts not working with non ENG keyboard layout

### ISSUE THREAD

**Issue by romanraspopov** [camunda/camunda-modeler#2724]
Camunda Modeler keyboard shortcuts work fine when the keyboard layout is set to ENG.

![image](https://user-images.githubusercontent.com/4521867/152127865-d4795e3d-8bd6-4b5a-a9ea-218743122bbd.png)

Вut when I change keyboard layout to RUS, shortcuts stop working.
I have to switch the keyboard layout all the time or use menu Edit and select Undo, Redo and so on which is timewasting.
It would be great to fix it in order to help users save time.

__UPD__:
One-symbol shortcuts (like "H", "L" and others) work fine with any keyboard layout. Only composite shortcuts ("Ctrl + C", etc.) crash.
![image](https://user-images.githubusercontent.com/4521867/152160900-67498f83-4685-4ebf-afde-5e5ee6c289c7.png)

__Expected Behavior__

Keyboard shortcuts should work with any keyboard layout.

__Environment__

 * Version: 4.12.0
 * Operating System: Windows 10 amd64
 * Plugins: <no plugins>
 * Execution Platform: BPMN - Camunda Platform

**Comment by nikku** [comment]
Thanks for opening this bug.

I can see how annoying that is. How do other apps usually handle Russian keyboard layout? I.e. how does VSCode keyboard shortcuts change between English and Russian? Is it still the same button to be pressed (on the keyboard) or is it the same / similar key?

**Comment by romanraspopov** [comment]
Thanks for reply!
I`ve checked VSCode, it handles Russian keyboard layout very well. All shortcuts work fine.

**Comment by kerlon5** [comment]
@romanraspopov @nikku just cheked it on windows 10 and ubuntu 22.10 with russian keyboard - can't reproduce

**Comment by romanraspopov** [comment]
I`ve just updated the version to 5.4.2 and checked it on Windows 10. The
bug remains. I can`t use Ctrl+C and Ctrl+V keyboard shortcuts when the
Russian layout is set. I have to switch to English.

вс, 23 окт. 2022 г. в 15:40, Victor Naumov ***@***.***>:

**Comment by kerlon5** [comment]
double checked this on windows 10 with Russian as system language - issue confirmed. on windows 10 where Russian is only second input language this problem does not exist. @romanraspopov do you have Russian as system language?

**Comment by romanraspopov** [comment]
Yes, Russian is set as system language (default). I can use only menu
Copy/Cut/Paste commands when Russian layout is set. Keyboard shortcuts
Ctrl+C,  Ctrl+X and Ctrl+V don`t work. However the other shortcuts (Ctrl+Z,
Ctrl+Y, H, L, S, etc.) work fine.

вт, 25 окт. 2022 г. в 10:46, Victor Naumov ***@***.***>:

**Comment by kerlon5** [comment]
@barmac I'm not sure that you can close this without merging #3264

**Comment by barmac** [comment]
I agree. This issue was in fact closed automatically due to this fragment in the PR:
> fix https://github.com/camunda/camunda-modeler/issues/2724

**Comment by nikku** [comment]
@romanraspopov @kerlon5 As we investigate a fix, just to double check: 

* You are on Windows
* You use a cyrillic keyboard layout
* You expect which keys exactly to trigger undo / redo? These that bind to the physical `KeyZ` and `KeyY`?

**Comment by romanraspopov** [comment]
Looks like I`ve got some software conflict on my laptop. There is nothing
to do with camunda, the problem is on my side.

вт, 14 февр. 2023 г. в 20:18, Nico Rehwaldt ***@***.***>:

**Comment by nikku** [comment]
As we do it we need to ensure to not break existing keyboard layouts (i.e. German QWERTZ) with the addition of physical keys.

Physical keys must always be accounted for _after_ actual keys, I think. If we want to go the alternative route we need to allow users to _explicitly_ configure keyboard bindings. That is the route most editors (VSCode, ...) do to support non latin keyboards, i.e. Russian.

Alternatively we could attempt to detect keyboard layouts via [`Keyboard#getLayoutMap`](https://developer.mozilla.org/en-US/docs/Web/API/Keyboard/getLayoutMap) and conditionally enable physical keys.
